agents-api/agents_api/queries/agents/update_agent.py

```diff
@@ -14,16 +14,40 @@
 
 # Define the raw SQL query
 agent_query = """
-UPDATE agents
-SET
-    metadata = $3,
-    name = $4,
-    about = $5,
-    model = $6,
-    default_settings = $7::jsonb,
-    default_system_template = $8
-WHERE agent_id = $2 AND developer_id = $1
-RETURNING *;
+WITH proj AS (
```

Comment:
this query is rather suboptimal no? it should only be checking the project iff it's being changed at all